chain/epoch_manager/src/lib.rs

```diff
@@ -627,8 +637,37 @@ impl EpochManager {
                 vec![prev_prev_stake, prev_stake, new_stake].into_iter().max().unwrap();
             stake_info.insert(account_id.to_string(), max_of_stakes);
         }
+        let slashing_info = self.compute_double_sign_slashing_info(last_block_hash)?;
         debug!(target: "epoch_manager", "stake_info: {:?}, validator_reward: {:?}", stake_info, validator_reward);
-        Ok((stake_info, validator_reward))
+        Ok((stake_info, validator_reward, slashing_info))
+    }
+
+    /// Compute slashing information. Returns a hashmap of account id to slashed amount for double sign
+    /// slashing.
+    fn compute_double_sign_slashing_info(
+        &mut self,
+        last_block_hash: &CryptoHash,
+    ) -> Result<HashMap<AccountId, Balance>, EpochError> {
+        let slashed = self.get_slashed_validators(last_block_hash)?.clone();
+        let epoch_id = self.get_epoch_id(last_block_hash)?;
+        let epoch_info = self.get_epoch_info(&epoch_id)?;
+        let total_stake: Balance = epoch_info.validators.iter().map(|v| v.amount).sum();
+        let mut res = HashMap::default();
+        for (account_id, slash_state) in slashed {
+            if let SlashState::DoubleSign = slash_state {
+                if let Some(&idx) = epoch_info.validator_to_index.get(&account_id) {
+                    let stake = epoch_info.validators[idx as usize].amount;
+                    let slashed_stake = if stake * 3 >= total_stake {
```

Comment:
This needs comment. Cause I don't understand the logic when you multiple `stake * stake`